### DISCUSSION TITLE
How do I initialize a StackedModel properly?

### DISCUSSION THREAD

**Discussion by VanBuren77** [biolab/orange3#6120] [General]
First off - great Library, huge fan! I'm very confused about proper model instantiation in Python for Stacked Models. It looks like there's a StackedFitter() included in one of the Python stacked files in the __main__ but it doesn't return the inner models correctly

When I pickle this model in Orange:

![image](https://user-images.githubusercontent.com/20358136/188199226-c89ce699-da9e-4e92-a4f9-14cf5fd6f709.png)

Load it as a pickled object into python, I see the following:

![image](https://user-images.githubusercontent.com/20358136/188199160-c9f38192-0c4b-43fc-a76e-42c5316fe402.png)

How can I re-create this init process in python?

I have the following code to attempt this:
`
def _get_model():
    n_learners = 1 # no dunder fields ->
    sub_stack_count = 1
    nets_per_sub_stack = 3

    target_variable = "target_log_9d"
    nets_per_stack_count = 3
    stack_count = 2
    hidden_layer_sizes = (5, 10, 10, 5)
    activation = 'tanh' # activation = 'relu'
    solver = 'adam'
    learning_rate = 0.01
    batch_sizes = "auto"
    alpha = 0.0006
    # aggregator = LinearRegression
    # aggregator = Orange.regression.LinearRegressionLearner()
    aggregator = LinearRegressionLearner
    warmup_period = 120
    max_iter = 500

    all_models = []
    aggregation_func = aggregator()

    # Set n top level MLPs: ->
    for i in range(1):
        master_net = NNRegressionLearner(hidden_layer_sizes, activation=activation, alpha=alpha, max_iter=max_iter)
        
        call_back = MLPRegressorWCallback(activation='logistic', alpha=0.0006,  
                                          hidden_layer_sizes=(5, 10, 10, 5), max_iter=500, random_state=1)

        master_net = SklModelRegression(skl_model=call_back)
        all_models.append(master_net)

    # Set n substacks: ->
    for i in range(sub_stack_count):
        sub_nets = []
        
        for j in range(nets_per_sub_stack):
            
            sub_net = NNRegressionLearner(hidden_layer_sizes, activation=activation,
                                          alpha=alpha, max_iter=max_iter)
             # Have tried both NNRegressionLearner() and MLPRegressorWcallback()!
            call_back = MLPRegressorWCallback(activation='logistic', 
                                              alpha=0.0006, 
                                              hidden_layer_sizes=(5, 10, 10, 5), 
                                              max_iter=500, 
                                              random_state=1)
            sub_net = SklModelRegression(skl_model=call_back)
            sub_nets.append(sub_net)
        sub_stack = StackedModel(models=sub_nets, aggregate=aggregation_func,use_prob=False)

    final_model = StackedModel(models=all_models, aggregate=aggregation_func, use_prob=False)
    return final_model
    # SklModelRegression(skl_model=MLPRegressorWCallback(activation='logistic', alpha=0.0006,
                #   hidden_layer_sizes=(5, 10, 10, 5), max_iter=500,
                #   random_state=1))  # params={'hidden_layer_sizes': (5, 10, 10, 5), 'activation': 'logistic', 'solver': 'adam', 'alpha': 0.0006, 'batch_size': 'auto', 'learning_rate': 'constant', 'learning_rate_init': 0.001, 'power_t': 0.5, 'max_iter': 500, 'shuffle': True, 'random_state': 1, 'tol': 0.0001, 'verbose': False, 'warm_start': False, 'momentum': 0.9, 'nesterovs_momentum': True, 'early_stopping': False, 'validation_fraction': 0.1, 'beta_1': 0.9, 'beta_2': 0.999, 'epsilon': 1e-08}
    # all_models.append(sub_stack)`

What is returned is not the object I pickled and loaded above

Thanks,
Evan
